### DISCUSSION TITLE
Tailwind @screen directive equivalent

### DISCUSSION THREAD

**Discussion by saeedPadyab** [ben-rogerson/twin.macro#179] [General] [answered]
Hello
I want to know how can I use tailwind directives with twin?

**Comment by ben-rogerson** [comment, marked as answer]
Which directives are you interested in using?

**Reply by saeedPadyab** [reply]
Thanks for your response. The most important which is @screen

**Reply by saeedPadyab** [reply]
it seems i can have something like this.
`const Wrapper = styled.div`
  ${theme`screens.xl`} {
    ${tw`bg-red-500`}
  }`
`

**Comment by ben-rogerson** [comment]
You could use the tw import for that example:

```
const Wrapper = tw.div`xl:bg-red-500`;
```

But if you're writing conditional styles or regular css then you'll need to use the styled import (or css prop).

If you want to work with the config breakpoints, you could make reusable functions to help sync them with tailwind:

```js
import tw, { css, theme } from 'twin.macro'

const minXl = content => css`
  @media (min-width: ${theme`screens.xl`}) {
    ${content}
  }
`

const xlStyles = minXl(`
  ${tw`block`}
  color: red;
`)

const StyledButton = styled.button(() => [
    tw`bg-black`,
    xlStyles,
])
```

As of 1.12.0, the theme import allows you to grab the screens object from the config (as opposed to a single value). 
This means it's now possible to map over screens to make a dynamic breakpoint helper.

**Reply by saeedPadyab** [reply]
Thanks for the response, I did exactly the way you said.
